test/functional/mining_basic.py

```diff
@@ -237,10 +237,17 @@ def assert_submitblock(block, result_str_1, result_str_2=None):
         bad_block = copy.deepcopy(block)
         bad_block.vtx[0].vin[0].prevout.hash += 1
         bad_block.vtx[0].rehash()
+        self.log.info("getblocktemplate: Test block with missing coinbase")
         assert_template(node, bad_block, 'bad-cb-missing')
 
+        self.log.info("submitblock: Test empty block")
+        assert_equal('high-hash', node.submitblock(hexdata=CBlock().serialize().hex()))
         self.log.info("submitblock: Test invalid coinbase transaction")
-        assert_raises_rpc_error(-22, "Block does not start with a coinbase", node.submitblock, CBlock().serialize().hex())
+        bad_block.solve()
+        assert_raises_rpc_error(-22, "Block does not start with a coinbase", node.submitblock, bad_block.serialize().hex())
+        no_tx_block = copy.deepcopy(block)
+        no_tx_block.vtx.clear()
+        no_tx_block.solve()
         assert_raises_rpc_error(-22, "Block does not start with a coinbase", node.submitblock, bad_block.serialize().hex())
```

Comment:
submit `no_tx_block` instead of `bad_block` here, and if I remember correctly from yesterday, that will lead to a different error.